packages/components/autocomplete/src/use-autocomplete.ts

```diff
@@ -339,15 +339,26 @@ export function useAutocomplete<T extends object>(originalProps: UseAutocomplete
     }
   }, [inputRef.current]);
 
-  // focus first non-disabled item
+  // focus selected item or first non-disabled item
   useEffect(() => {
-    let key = state.collection.getFirstKey();
-
-    while (key && state.disabledKeys.has(key)) {
-      key = state.collection.getKeyAfter(key);
+    let keyToFocus;
+
+    if (
+      state.selectedKey != null &&
```

Comment:
```tsx
state.selectedKey !== null
```

Reply:
- thank you for suggestion 😄  (updated: https://github.com/heroui-inc/heroui/pull/5290/commits/eacecb76462fbe8e50dc5aad46db480d8f6c0cf0)